src/safeposix/syscalls/net_calls.rs

```diff
@@ -1571,8 +1887,13 @@ impl Cage {
         flags: i32,
         addr: &mut Option<&mut interface::GenSockaddr>,
     ) -> i32 {
-        // maybe select reported a INPROGRESS tcp socket as readable, so re-check the
-        // state here
+        //In the case that the socket is nonblocking and the connection can not
+        //be completed immediately, the connection state of the socket will
+        //be set to INPROGRESS.
+        //It is possible that select_syscall or poll_syscall had reported
+        //the INPROGRESS TCP socket as readable. If so, we can adjust the
+        //state of the socket to CONNECTED.
+        //** In what case would this happen?? */
```

Comment:
INPROGRESS state may happen when you try to connect with nonblocking flag, which I think is a half-broken feature right now. Logics related to INPROGRESS state in select is also a mess currently